### PR TITLE
Migrate to antlr lexer/parser

### PR DESCRIPTION
**Pull request by ingwarsw**
That PR is just "partially" working attempt to migrate to antlr lexer/parser.
Its not working because I was not able to make lexer working properly.

I opened it to have idea" how to make it in the future..
Only relevant commits here are https://github.com/earthly/earthly-intellij-plugin/pull/35/commits/f78f2b319e435844156caec1c630c058a7150015 and https://github.com/earthly/earthly-intellij-plugin/pull/35/commits/8c312522502e03e97ea0e1a185136b379b60532e

Just for the future..
Lexer/parser was working except for.

IDENT/DEDENT setup..
Some random errors `Discontinuous sequence of tokens is generated by lexer`

### CODE REVIEW

**Comment by idelvall** [comment]
Thanks @ingwarsw! 
give me a couple of days to take a look

File: Earthfile [hunk 1]
```diff
@@ -2,7 +2,8 @@ VERSION --pass-args --global-cache --use-function-keyword 0.7
 PROJECT earthly-technologies/core
 IMPORT github.com/earthly/lib/gradle:0a891b93330eced8fe57f48397a1109d829cb7d4 AS gradle
 ARG --global gradle_version=8.2.1
-ARG --global bundle="github.com/earthly/earthfile-grammar+export/"
+ARG --global bundle="../earthfile-grammar+export/"
+#ARG --global bundle="github.com/earthly/earthfile-grammar+export/"
```
Comment (idodod) [inline Earthfile:6]: undo?

File: src/main/antlr/EarthLexer.g4 [hunk 2]
```diff
@@ -0,0 +1,231 @@
+lexer grammar EarthLexer;
```
Comment (idodod) [inline src/main/antlr/EarthLexer.g4:1]: is there a way for us to still use the same lexer defined in earthly/earthly (or somehow use it as source of truth)?

File: src/main/java/dev/earthly/plugin/language/syntax/EarthlyLineMarkerProvider.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+import java.util.Collection;
+import java.util.List;
+
+final class EarthlyLineMarkerProvider extends RelatedItemLineMarkerProvider {
+
+    @Override
+    protected void collectNavigationMarkers(@NotNull PsiElement element,
+                                            @NotNull Collection<? super RelatedItemLineMarkerInfo<?>> result) {
+        System.out.println("In marker: " + element);
```
Comment (idodod) [inline src/main/java/dev/earthly/plugin/language/syntax/EarthlyLineMarkerProvider.java:20]: should we remove the println statements?

File: src/main/java/dev/earthly/plugin/language/syntax/highlighting/EarthlyHighlighter.java [hunk 4]
```diff
@@ -27,53 +33,56 @@ public class EarthlyHighlighter extends SyntaxHighlighterBase {
...
-    COLOR_MAP.put("punctuation.definition.variable.begin.earthfile", PARENTHESES);
-    COLOR_MAP.put("punctuation.definition.variable.end.earthfile", PARENTHESES);
-    COLOR_MAP.put("punctuation.definition.string.begin.earthfile", STRING);
-    COLOR_MAP.put("punctuation.definition.string.end.earthfile", STRING);
-    COLOR_MAP.put("string.quoted.double.earthfile", STRING);
-    COLOR_MAP.put("string.quoted.single.earthfile", STRING);
-    COLOR_MAP.put("keyword.other.special-method.earthfile", KEYWORD);
-    COLOR_MAP.put("variable.other.earthfile", VARIABLE);
-  }
+//    COLOR_MAP.put("constant.numeric-port.earthfile", EXPOSE_PORT);
```
Comment (idodod) [inline src/main/java/dev/earthly/plugin/language/syntax/highlighting/EarthlyHighlighter.java:39]: should we delete these commented lines?

File: src/main/java/dev/earthly/plugin/language/syntax/EarthlyLineMarkerProvider.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+//             Add the property to a collection of line marker info
+            NavigationGutterIconBuilder<PsiElement> builder =
+                    NavigationGutterIconBuilder.create(EarthlyIcons.FILE)
+                            .setTargets(functions)
+                            .setTooltipText("Navigate to Earthly language property");
+            result.add(builder.createLineMarkerInfo(element));
+        }
+    }
+
+}
```
Comment (idodod) [inline src/main/java/dev/earthly/plugin/language/syntax/EarthlyLineMarkerProvider.java:41]: can we add newlines here and in other files they are missing?

**Review by idodod — COMMENTED** [review]
Thanks @ingwarsw for the contribution!
I'll defer to @idelvall for final approval but just left a few comments in the meantime.

**Comment by ingwarsw** [comment]
@idelvall @idodod Thats bad PR.. that one is not working and its just for future use..
Proper one is #34

**Pull request closed without merging**